esmvaltool/preprocessor/_regrid.py

```diff
@@ -173,8 +190,30 @@ def regrid(cube, target_grid, scheme):
         else:
             # Generate a target grid from the provided cell-specification,
             # and cache the resulting stock cube for later use.
-            target_grid = _CACHE.setdefault(target_grid,
-                                            _stock_cube(target_grid))
+            if not lon_offset and lat_offset:
```

Comment:
This code is unnessarily complicated, replace by:
```python
            target_grid = _CACHE.setdefault(
                target_grid,
                _stock_cube(target_grid, lat_offset, lon_offset),
            )
```

Reply:
not as easy as it seems, this is complicated but preserves the positional arguments structure for `_regrid()` that otherwise would have to be altered and we'd have to change them everywhere we use the function

Reply:
Could you explain why? I don't think I understand what you mean.

Reply:
have a look at the unit test for regrid ie l.120:
```
result = regrid(self.src_cube, spec, scheme)
```
this will fail with a mock `TypeError` error alike
```
TypeError: <lambda>() takes 1 positional argument but 3 were given
```

Reply:
Just update the test so it corresponds to the new situation

Reply:
no the test should not be updated, the test preserves the default number of positional arguments for regrid: cube, target, scheme and should be kept this way since regrid is called in a lot of places like this. If you call stock_cube with (target_grid, lat_offset, lon_offset) inside regrid() this will result in `TypeError: <lambda>() takes 1 positional argument but 3 were given` because lat_offset and lon_offset are keyword arguments so the error is correct; these are set by default to True in stock_cube so if the user is not specifying them they are set automatically and they should not be used in the function call; if the user sets them then the function call should be updated accordingly - hence my convoluted checks that are actually needed. If you find a better way then I am going to buy you a beer next time we meet :beer:

Reply:
I've updated the call signature to reflect that of the new version of `_stock_cube` and now it works fine. I've pushed the changes. You owe me a :beer: mister ;-)

Reply:
very nice one @bouweandela - I didn't even see that `side_effect` function! Indeed I do owe you a beer :beers: